packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -274,36 +271,54 @@ class Auth {
+     * is not found in local store or if SFRA wants PWA to trigger refresh token login.
+     *
+     * On PWA-only sites, this returns the access token from local storage.
+     * On Hybrid sites, this checks whether SFRA has sent an auth token via cookies.
+     * Returns an access token from SFRA if it exist.
+     * If not, the access token from local store is returned.
+     *
+     * This is only used within this Auth module since other modules consider the access
+     * token from this.get('access_token') to be the source of truth.
+     *
+     * @returns {string} access token
      */
-    private hasSFRAAuthStateChanged() {
-        const refreshTokenKey =
-            (this.get('refresh_token_registered') && 'refresh_token_registered') ||
-            'refresh_token_guest'
+    private getAccessToken() {
+        let accessToken = this.get('access_token')
+        const sfraAuthToken = this.get('access_token_sfra')
+
+        if (sfraAuthToken && accessToken !== sfraAuthToken) {
+            /*
+             * If SFRA sends 'refresh', we return an empty token here so PWA can trigger a login refresh
+             * This key is used when logout is triggered in SFRA but the redirect after logout
+             * sends the user to PWA.
+             */
+            if (sfraAuthToken === 'refresh') {
+                this.set('access_token', '')
+                this.clearSFRAAuthToken()
+                return ''
+            }
 
-        const refreshTokenCopyKey =
-            (this.get('refresh_token_registered_copy') && 'refresh_token_registered_copy') ||
-            'refresh_token_guest_copy'
+            const {isGuest, customerId, usid} = this.parseSlasJWT(sfraAuthToken)
+            this.set('access_token', sfraAuthToken)
+            this.set('customer_id', customerId)
+            this.set('usid', usid)
+            this.set('customer_type', isGuest ? 'guest' : 'registered')
 
-        if (DATA_MAP[refreshTokenKey].key !== DATA_MAP[refreshTokenCopyKey].key) {
-            return true
+            accessToken = sfraAuthToken
+            // SFRA -> PWA access token cookie handoff is succesful so we clear the SFRA made cookies.
+            // We don't want these cookies to persist and continue overriding what is in local store.
+            this.clearSFRAAuthToken()
         }
 
-        return this.get(refreshTokenKey) !== this.get(refreshTokenCopyKey)
+        return accessToken
     }
 
-    /**
-     * Used to validate JWT expiry and ensure auth state consistency with SFRA in a hybrid setup
-     * @param token access_token received on SLAS authentication
-     * @returns {boolean} true if JWT is valid; false otherwise
-     */
-    private isTokenValidForHybrid(token: string) {
-        return !this.isTokenExpired(token) && !this.hasSFRAAuthStateChanged()
+    private clearSFRAAuthToken() {
+        const {key, storageType} = DATA_MAP['access_token_sfra']
```

Comment:
QUESTION: I know there are now references to "srfa" in our code.. I want to make sure that we don't have any references that are on non-private functions/variables. If there is we should have a discussion about naming.. e.g. do we refer to it as a external system token? This isn't a big deal, but I just don't want to end up having public API's on the commerce sdk react with reference to SRFA

Reply:
This is a good point. But in this case, the SFRA references are only on private functions and variables that only auth/index.ts and its associated test knows about. The externally visible part of this change is the `cc-at` name (short for cc-accesstoken, with the `cc-` namespace being the same as what we use for the refresh token and other plugin_slas cookies) that we are using for sending the access token cookie.